hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,55 @@
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def migration
+  export_dir = '/var/tmp/engine_upgrade'
+  mongo_dir = '/var/lib/mongodb'
+  hiera_file = '/etc/foreman-installer/custom-hiera.yaml'
+  mongo_backup = '/var/tmp/mongo_backup'
+  mongo_conf = '/etc/opt/rh/rh-mongodb34/mongod.conf'
+
+  # Verify we have a good value in the answerfile for the MongoDB DB name.
+  pulp_db_param = param('katello', 'pulp_db_name') || param('foreman_proxy_content', 'pulp_db_name')
+  fail_and_exit('Unknown pulp database name') unless pulp_db_param
+  mongo_db_name = pulp_db_param.value
+
+  # Create export directory and dump MongoDB
+  logger.info 'Stopping Pulp services except MongoDB'
+  Kafo::Helpers.execute('katello-service stop --exclude rh-mongodb34-mongod')
+  FileUtils.mkdir_p(export_dir)
```

Comment:
I'm debating whether `mkdir_p` is a good idea. If the directory already exists, isn't that reason for concern? I'm not familiar with mongodb so I don't know what happens if there's already a dump there.

Reply:
Good catch, fixed to just mkdir